crates/router/src/core/admin.rs

```diff
@@ -219,6 +224,14 @@ pub async fn merchant_account_update(
         locker_id: req
             .locker_id
             .or_else(|| merchant_account.locker_id.to_owned()),
+        metadata: if req.metadata.is_some() {
+            Some(
+                utils::Encode::<api::MerchantDetails>::encode_to_value(&req.metadata)
+                    .change_context(errors::ApiErrorResponse::InternalServerError)?,
+            )
+        } else {
+            merchant_account.metadata.to_owned()
+        },
```

Comment:
Use `and_then().or()`:

```suggestion
        metadata: 
            req.metadata.map(|metadata|
                utils::Encode::<api::MerchantDetails>::encode_to_value(metadata)
                    .change_context(errors::ApiErrorResponse::InternalServerError)
            )
            .unwrap_or(Ok(merchant_account.metadata))?
```